gradle-metric-schema/src/main/java/com/palantir/metric/schema/gradle/MetricSchemaPlugin.java

```diff
@@ -122,17 +124,18 @@ private static void configureJavaSource(Project project, Provider<Directory> out
 
     private static void configureEclipse(Project project, TaskProvider<? extends Task> generateMetrics) {
         project.getPluginManager().withPlugin("eclipse", _plugin -> {
-            Task task = project.getTasks().findByName("eclipseClasspath");
-            if (task != null) {
-                task.dependsOn(generateMetrics);
+            try {
+                project.getTasks().named("eclipseClasspath").configure(t -> t.dependsOn(generateMetrics));
+            } catch (UnknownDomainObjectException e) {
```

Comment:
Might be a bit clearer to use [`UnknownTaskException`](https://docs.gradle.org/current/javadoc/org/gradle/api/UnknownTaskException.html) as that's what's explicitly called out in the javadoc.